bugsnag/utils.py

```diff
@@ -118,7 +118,19 @@ def _sanitize_dict(self, obj, trim_strings, ignored):
 
         clean_dict = {}
         for key, value in six.iteritems(obj):
-            clean_dict[key] = self._sanitize(value, trim_strings, ignored)
+
+            clean_value = self._sanitize(value, trim_strings, ignored)
+
+            if isinstance(key, basestring):
+                clean_dict[key] = clean_value
+            else:
+                try:
+                    clean_dict[str(key)] = clean_value
+                except:
+                    try:
+                        clean_dict[repr(key)] = clean_value
```

Comment:
We should be wary of using the `repr` function response as a key, it may produce a lot more information than was intended for use in this area.